modules/apps/portal-cache/portal-cache-test/src/testIntegration/java/com/liferay/portal/cache/internal/test/PortalCacheExtenderTest.java

```diff
@@ -0,0 +1,317 @@
+		_bundle.stop();
+
+		_bundle.update(
+			_createBundle(
+				"module-multi-vm-updated.xml", "module-single-vm-updated.xml"));
+
+		_bundle.start();
+
+		_assertCacheConfig(
+			PortalCacheManagerNames.MULTI_VM, false, 2001,
+			"test.cache.multi.updated", true, 101L);
+	}
+
+	@Test
+	public void testUpdateModuleSingleVMConfig() throws Exception {
+		_bundle.stop();
+
+		_bundle.update(
+			_createBundle(
+				"module-multi-vm-updated.xml", "module-single-vm-updated.xml"));
+
+		_bundle.start();
+
+		_assertCacheConfig(
+			PortalCacheManagerNames.SINGLE_VM, false, 2001,
+			"test.cache.single.updated", true, 101L);
+	}
+
+	private void _assertCacheConfig(
+			String cacheManagerName, boolean eternal, int maxElementsInMemory,
+			String name, boolean overflowToDisk, long timeToIdleSeconds)
+		throws Exception {
+
+		MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+		ObjectName objectName = new ObjectName(
+			StringBundler.concat(
+				"net.sf.ehcache:type=CacheConfiguration,CacheManager=",
+				cacheManagerName, ",name=", name));
+
+		Assert.assertEquals(
+			eternal, mBeanServer.getAttribute(objectName, "Eternal"));
+		Assert.assertEquals(
+			maxElementsInMemory,
+			mBeanServer.getAttribute(objectName, "MaxElementsInMemory"));
+		Assert.assertEquals(name, mBeanServer.getAttribute(objectName, "Name"));
+		Assert.assertEquals(
+			overflowToDisk,
+			mBeanServer.getAttribute(objectName, "OverflowToDisk"));
+		Assert.assertEquals(
+			timeToIdleSeconds,
+			mBeanServer.getAttribute(objectName, "TimeToIdleSeconds"));
+	}
+
+	private InputStream _createBundle(
+			String multiCacheConfigName, String singleCacheConfigName)
+		throws IOException {
+
+		try (UnsyncByteArrayOutputStream unsyncByteArrayOutputStream =
+				new UnsyncByteArrayOutputStream()) {
+
+			try (JarOutputStream jarOutputStream = new JarOutputStream(
+					unsyncByteArrayOutputStream)) {
+
+				_writeManifest(_BUNDLE_SYMBOLIC_NAME, "1.0.0", jarOutputStream);
+
+				_writeClass(jarOutputStream);
+
+				if (multiCacheConfigName != null) {
+					_writeResource(
+						jarOutputStream,
+						"/com/liferay/portal/cache/internal/test/" +
+							multiCacheConfigName,
+						"META-INF/module-multi-vm.xml");
+				}
+
+				if (singleCacheConfigName != null) {
+					_writeResource(
+						jarOutputStream,
+						"/com/liferay/portal/cache/internal/test/" +
+							singleCacheConfigName,
+						"META-INF/module-single-vm.xml");
+				}
+			}
+
+			return new UnsyncByteArrayInputStream(
+				unsyncByteArrayOutputStream.unsafeGetByteArray(), 0,
+				unsyncByteArrayOutputStream.size());
+		}
+	}
+
+	private Object _fetchMBeanObject(String cacheManagerName, String cacheName)
+		throws Exception {
+
+		MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+
+		ObjectName objectName = new ObjectName(
+			StringBundler.concat(
+				"net.sf.ehcache:type=CacheConfiguration,CacheManager=",
+				cacheManagerName, ",name=t", cacheName));
```

Comment:
@kyle-miho 
The "name=t" seems a typo to me.
So in your test logic, when the file existed, you verify if the cache is there, when the file did not exist, you verify the MBean, and all the verify for MBean is to verify its not there. This is dangerous as you would never know if you have anything wrong in your _fetchMBeanObject method logic.
You should add a checking to verify MBean not null when the file existed.

Tina.